test/models/name_test.rb

```diff
@@ -2503,6 +2533,80 @@ def test_other_approved_synonyms
     assert_empty(names(:lactarius_alpinus).other_approved_synonyms)
   end
 
+  def test_best_preferred_synonym
+    # no preferred synonyms
+    assert_empty(names(:pluteus_petasatus_deprecated).best_preferred_synonym)
+
+    # only 1 preferred synonym
+    assert_equal(names(:lactarius_alpinus),
+                 names(:lactarius_alpigenes).best_preferred_synonym)
+
+    # > 1 preferred synonym, none with observations
+    # Macrolepiota rachodes & rhacodes are synonyms, approved, and have
+    # no observations
+    # Create a deprecated synonym and test it
+    deprecated_name = Name.create!(
+      text_name: "Lepiota rhacodes",
+      author: "(Vittad.) Quél.",
+      display_name: "__Lepiota rhacodes__ (Vittad.) Quél.",
+      synonym: synonyms(:macrolepiota_rachodes_synonym),
+      deprecated: true,
+      rank: :Species, user: users(:rolf)
+    )
+    # M. rachodes & rhacodes are tied with 0 Observations
+    # "Best" one is the one last updated
+    assert_equal(names(:macrolepiota_rachodes),
+                 deprecated_name.best_preferred_synonym)
+
+    # > 1 preferred synonyms, one with observations
+    # C. rachodes is approved, has 1 Observation
+    # C. rachodes is approved, has 0 Observations
+    # Create a deprecated synonym and test it
+    deprecated_name = Name.create!(
+      text_name: "Agaricus rhacodes",
+      author: "Vittad.",
+      display_name: "__Agaricus rhacodes__ Vittad.",
+      synonym: synonyms(:chlorophyllum_rachodes_synonym),
+      deprecated: true,
+      rank: :Species, user: users(:rolf)
+    )
+    assert_equal(names(:chlorophyllum_rachodes),
+                 deprecated_name.best_preferred_synonym)
+
+    # > 1 preferred synonyms, > 1 with observations,
+    # Neither has more Observations
+    # Create an Observation for the other approved synonym, so that they'll
+    # be tied in # of Observations
+    revised_best_synonym = names(:chlorophyllum_rhacodes)
+    Observation.create(
+      name: revised_best_synonym,
+      user: users(:rolf), when: Time.current, location: locations(:albion)
+    )
+    # other_approved_synonyms.name.observations is cached by Rails, so
+    # it didn't change when we created the Observation above.
+    # So reload it
+    deprecated_name.other_approved_synonyms.
```

Comment:
Before I added the `reload`, I was getting mysterious behavior which completely stumped me. @pellaea talked me through it.
I hope the code comments explain the `reload` adequately.